In [1]:
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt
import seaborn as sns 
import plotly.graph_objects as go


In [2]:
world = pd.read_csv("owid-covid-data.csv", sep = ",")

In [3]:
world.isnull().sum()


iso_code                                   346
continent                                  708
location                                     0
date                                         0
total_cases                                585
new_cases                                  592
new_cases_smoothed                        1547
total_deaths                              9048
new_deaths                                9049
new_deaths_smoothed                       1547
total_cases_per_million                    931
new_cases_per_million                      938
new_cases_smoothed_per_million            1888
total_deaths_per_million                  9381
new_deaths_per_million                    9382
new_deaths_smoothed_per_million           1888
reproduction_rate                        11938
icu_patients                             54154
icu_patients_per_million                 54154
hosp_patients                            53103
hosp_patients_per_million                53103
weekly_icu_ad

C6. Show different tendencies for each column (top 10 with less NaNs) in your dataset.
Show, vertically, the start date and end date of the alarm state in each plot. If there
is no alarm state, then show only the start date.


B4. When are the worst moments to go to the countries? Answer this referring to the
“per_million” columns.


Extraigo las columnas necesarias para el análisis de tendencias por columnas y el posterior de peor momento para ir a los paises (columnas _per_million)

Analisis de tendencias por columnas: columnas de estudio: Descarto las weekly_ por la cantidad de nulos. Los paises europeos tienen mejor calidad de datos que Turquia y Venezuela, cuya cantidad de NaNs me hace descartar algunas columnas mas.

Elimino las que tienen muchos valores nulos 

icu_patients                             54154
icu_patients_per_million                 54154
hosp_patients                            53103
hosp_patients_per_million                53103
weekly_icu_admissions                    60119
weekly_icu_admissions_per_million        60119
weekly_hosp_admissions                   59743
weekly_hosp_admissions_per_million       59743
new_tests                                32672
total_tests                              32853
total_tests_per_thousand                 32853
new_tests_per_thousand                   32672
new_tests_smoothed                       29207
new_tests_smoothed_per_thousand          29207
tests_per_case                           31440
tests_units                              28050
total_vaccinations                       60083
new_vaccinations                         60252
new_vaccinations_smoothed                59872
total_vaccinations_per_hundred           60083
new_vaccinations_smoothed_per_million    59872
female_smokers                           14609
male_smokers                             15240
handwashing_facilities                   31473
extreme_poverty                          21146

Descarto las columnas con datos estimados (_smoothed), continent y location

Descarto las columnas que tienen un valor fijo o que no proporcionan unformación de tendencias.
median_age                                2949
aged_65_older                             3619
aged_70_older                             3276
gdp_per_capita                            2920
cardiovasc_death_rate                     2319
diabetes_prevalence                       1644
hospital_beds_per_thousand                7063
life_expectancy                            656
human_development_index                   2662

#Columnas en el DataFrame de estudio de tendencias

iso_code                                   346
date                                         0
total_cases                                585
new_cases                                  592
total_deaths                              9048
new_deaths                                9049
reproduction_rate                        11938
stringency_index                          6777
positive_rate                            30965 Uso esta columna a pesar de su alto valor de nulos
total_cases_per_million                    931
new_cases_per_million                      938
total_deaths_per_million                  9381
new_deaths_per_million                    9382
icu_patients_per_million                 54154
hosp_patients_per_million                53103


In [4]:
world_cols = world.drop(world.columns.difference(['iso_code', 'date', 'total_cases', 'new_cases', 'total_deaths', 'new_deaths', 'total_cases_per_million','new_cases_per_million','total_deaths_per_million', 'new_deaths_per_million','reproduction_rate', 'icu_patients_per_million', 'hosp_patients_per_million', 'positive_rate', 'stringency_index']), 1)

In [5]:
world_cols.columns

Index(['iso_code', 'date', 'total_cases', 'new_cases', 'total_deaths',
       'new_deaths', 'total_cases_per_million', 'new_cases_per_million',
       'total_deaths_per_million', 'new_deaths_per_million',
       'reproduction_rate', 'icu_patients_per_million',
       'hosp_patients_per_million', 'positive_rate', 'stringency_index'],
      dtype='object')

In [6]:
world_cols.isnull().sum()

iso_code                       346
date                             0
total_cases                    585
new_cases                      592
total_deaths                  9048
new_deaths                    9049
total_cases_per_million        931
new_cases_per_million          938
total_deaths_per_million      9381
new_deaths_per_million        9382
reproduction_rate            11938
icu_patients_per_million     54154
hosp_patients_per_million    53103
positive_rate                30965
stringency_index              6777
dtype: int64

In [7]:
gbr = world_cols[world_cols["iso_code"]=="GBR"]
prt = world_cols[world_cols["iso_code"]=="PRT"]
ven = world_cols[world_cols["iso_code"]=="VEN"]
tur = world_cols[world_cols["iso_code"]=="TUR"]
esp = world_cols[world_cols["iso_code"]=="ESP"]

In [8]:
gbr.dropna(how = 'any', inplace = True)
esp.dropna(how = 'any', inplace = True)
prt.dropna(how = 'any', inplace = True)
tur.dropna(how = 'any', inplace = True)
ven.dropna(how = 'any', inplace = True)

In [19]:
#VENEZUELA ME ESTA DANDO PROBLEMAS PORQUE HAY MUCHOS NULOS EN ESTAS COLUMNAS. TENGO QUE REVISAR SI LO HE HECHO BIEN
ven

Empty DataFrame
Columns: [iso_code, date, total_cases, new_cases, total_deaths, new_deaths, total_cases_per_million, new_cases_per_million, total_deaths_per_million, new_deaths_per_million, reproduction_rate, icu_patients_per_million, hosp_patients_per_million, positive_rate, stringency_index]
Index: []

In [9]:
# Columnas de estudio de los 5 paises analizados
countries_cols = pd.concat([gbr, prt, ven, tur, esp])
countries_cols.columns
countries_cols.isnull().sum()

iso_code                     0
date                         0
total_cases                  0
new_cases                    0
total_deaths                 0
new_deaths                   0
total_cases_per_million      0
new_cases_per_million        0
total_deaths_per_million     0
new_deaths_per_million       0
reproduction_rate            0
icu_patients_per_million     0
hosp_patients_per_million    0
positive_rate                0
stringency_index             0
dtype: int64

In [71]:
countries_cols.dtypes

iso_code                      object
date                          object
total_cases                  float64
new_cases                    float64
total_deaths                 float64
new_deaths                   float64
total_cases_per_million      float64
new_cases_per_million        float64
total_deaths_per_million     float64
new_deaths_per_million       float64
reproduction_rate            float64
icu_patients_per_million     float64
hosp_patients_per_million    float64
positive_rate                float64
stringency_index             float64
dtype: object

In [23]:
##?????
# world['date'] = pd.to_datetime(world['date'])
world['date']

0       2020-02-24
1       2020-02-25
2       2020-02-26
3       2020-02-27
4       2020-02-28
           ...    
60689   2021-01-13
60690   2021-01-14
60691   2021-01-15
60692   2021-01-16
60693   2021-01-17
Name: date, Length: 60694, dtype: datetime64[ns]

# GRAFICAS POR COLUMNAS

In [10]:
countries_cols.columns



Index(['iso_code', 'date', 'total_cases', 'new_cases', 'total_deaths',
       'new_deaths', 'total_cases_per_million', 'new_cases_per_million',
       'total_deaths_per_million', 'new_deaths_per_million',
       'reproduction_rate', 'icu_patients_per_million',
       'hosp_patients_per_million', 'positive_rate', 'stringency_index'],
      dtype='object')

In [46]:
fig = go.Figure(data=[go.Bar(x=countries_cols['date'], y=countries_cols['total_cases_per_million'])], layout_title_text="total_cases_per_million")
fig.show()

In [41]:

fig = go.Figure(data=[go.Bar(x=countries_cols['date'], y=countries_cols['new_cases_per_million'])], layout_title_text="new_cases_per_million")
fig.show()

In [33]:

fig = go.Figure(data=[go.Bar(x=esp['date'], y=esp['total_deaths_per_million'])], layout_title_text="total_deaths_per_million")
fig.show()

In [47]:
fig = go.Figure(data=[go.Bar(x=countries_cols['date'], y=countries_cols['new_deaths_per_million'])], layout_title_text="new_deaths_per_million")
fig.show()

In [49]:
fig = go.Figure(data=[go.Bar(x=countries_cols['date'], y=countries_cols['icu_patients_per_million'])], layout_title_text="icu_patients_per_million")
fig.show()

In [50]:
fig = go.Figure(data=[go.Bar(x=countries_cols['date'], y=countries_cols['hosp_patients_per_million'])], layout_title_text="hosp_patients_per_million")
fig.show()

In [11]:
countries_cols.describe().max()

total_cases                  2966244.000
new_cases                      68192.000
total_deaths                   79965.000
new_deaths                      1623.000
total_cases_per_million        45770.561
new_cases_per_million           1176.757
total_deaths_per_million        1177.931
new_deaths_per_million           671.000
reproduction_rate                671.000
icu_patients_per_million         671.000
hosp_patients_per_million        671.000
positive_rate                    671.000
stringency_index                 671.000
dtype: float64

In [14]:
esp.describe().max()
#gbr.describe().max()
#prt.describe().max()
#tur.describe().max()
ven.describe().max()

total_cases                  0.0
new_cases                    0.0
total_deaths                 0.0
new_deaths                   0.0
total_cases_per_million      0.0
new_cases_per_million        0.0
total_deaths_per_million     0.0
new_deaths_per_million       0.0
reproduction_rate            0.0
icu_patients_per_million     0.0
hosp_patients_per_million    0.0
positive_rate                0.0
stringency_index             0.0
dtype: float64

ANALISIS DEL PEOR MOMENTO PARA IR CON LAS COLUMNAS DESGINADAS CON PER_MILLION 

In [87]:
"""PER MILLION"""
countries_cols.loc[countries_cols["total_cases_per_million"] == countries_cols["total_cases_per_million"].max()]


iso_code        date  total_cases  new_cases  total_deaths  new_deaths  \
43910      PRT  2021-01-08     466709.0    10176.0        7590.0       118.0   

       total_cases_per_million  new_cases_per_million  \
43910                45770.561                997.969   

       total_deaths_per_million  new_deaths_per_million  reproduction_rate  \
43910                   744.358                  11.572               1.52   

       icu_patients_per_million  hosp_patients_per_million  positive_rate  \
43910                    52.566                    338.443          0.144   

       stringency_index  
43910             67.59

In [88]:
countries_cols.loc[countries_cols["new_cases_per_million"] == countries_cols["new_cases_per_million"].max()]


iso_code        date  total_cases  new_cases  total_deaths  new_deaths  \
51113      ESP  2020-11-02    1240697.0    55019.0       36257.0       379.0   

       total_cases_per_million  new_cases_per_million  \
51113                26536.258               1176.757   

       total_deaths_per_million  new_deaths_per_million  reproduction_rate  \
51113                   775.471                   8.106               1.15   

       icu_patients_per_million  hosp_patients_per_million  positive_rate  \
51113                    56.679                    410.012          0.121   

       stringency_index  
51113              71.3

In [90]:
countries_cols.loc[countries_cols["total_deaths_per_million"] == countries_cols["total_deaths_per_million"].max()]


iso_code        date  total_cases  new_cases  total_deaths  new_deaths  \
57389      GBR  2021-01-08    2966244.0    68192.0       79965.0      1333.0   

       total_cases_per_million  new_cases_per_million  \
57389                43694.485               1004.507   

       total_deaths_per_million  new_deaths_per_million  reproduction_rate  \
57389                  1177.931                  19.636               1.25   

       icu_patients_per_million  hosp_patients_per_million  positive_rate  \
57389                    45.635                    496.155          0.112   

       stringency_index  
57389             79.63

In [91]:
countries_cols.loc[countries_cols["new_deaths_per_million"] == countries_cols["new_deaths_per_million"].max()]


iso_code        date  total_cases  new_cases  total_deaths  new_deaths  \
51115      ESP  2020-11-04    1284408.0    25042.0       38118.0      1623.0   

       total_cases_per_million  new_cases_per_million  \
51115                27471.157                535.603   

       total_deaths_per_million  new_deaths_per_million  reproduction_rate  \
51115                   815.275                  34.713               1.12   

       icu_patients_per_million  hosp_patients_per_million  positive_rate  \
51115                    59.587                    434.715          0.127   

       stringency_index  
51115              71.3

In [92]:
countries_cols.loc[countries_cols["icu_patients_per_million"] == countries_cols["icu_patients_per_million"].max()]


iso_code        date  total_cases  new_cases  total_deaths  new_deaths  \
51127      ESP  2020-11-16    1496864.0    38273.0       41253.0       484.0   

       total_cases_per_million  new_cases_per_million  \
51127                32015.206                 818.59   

       total_deaths_per_million  new_deaths_per_million  reproduction_rate  \
51127                   882.327                  10.352                1.0   

       icu_patients_per_million  hosp_patients_per_million  positive_rate  \
51127                    67.501                    437.431          0.104   

       stringency_index  
51127              71.3

In [94]:
countries_cols.loc[countries_cols["hosp_patients_per_million"] == countries_cols["hosp_patients_per_million"].max()]

iso_code        date  total_cases  new_cases  total_deaths  new_deaths  \
57389      GBR  2021-01-08    2966244.0    68192.0       79965.0      1333.0   

       total_cases_per_million  new_cases_per_million  \
57389                43694.485               1004.507   

       total_deaths_per_million  new_deaths_per_million  reproduction_rate  \
57389                  1177.931                  19.636               1.25   

       icu_patients_per_million  hosp_patients_per_million  positive_rate  \
57389                    45.635                    496.155          0.112   

       stringency_index  
57389             79.63